internal/kargo-render/render.go

```diff
@@ -57,13 +62,19 @@ func RenderManifests(req Request) error { // nolint: revive
 		req.LocalInPath,
 		"--local-out-path",
 		req.LocalOutPath,
+		"--repo-username",
+		req.RepoCreds.Username,
 		"--output",
 		"json",
 	}
 	for _, image := range req.Images {
 		cmdTokens = append(cmdTokens, "--image", image)
 	}
 	cmd := exec.Command(cmdTokens[0], cmdTokens[1:]...) // nolint: gosec
+	cmd.Env = append(
+		os.Environ(),
+		fmt.Sprintf("KARGO_RENDER_REPO_PASSWORD=%s", req.RepoCreds.Password),
+	)
```

Comment:
One of two key changes. Most everything else is just updates to function signatures to get these creds down to here.